solutions/usaco-1114.mdx

```diff
@@ -37,37 +37,88 @@ $$
 ```cpp
 //CodeSnip{CPP Short Template}
 
-const int MAX_N = 300;
+const int MAX_N = 301;
 
 int A[MAX_N], dp[MAX_N][MAX_N];
 
 int main() {
 	setIO();
 	
 	int n; cin >> n;
-	for (int i = 0; i < n; i++) cin >> A[i];
-	for (int i = 0; i <= n; i++) {
-		for (int j = 0; j <= n; j++) {
+
+	for(int i = 0; i < n; i++) cin >> A[i];
+
+	for(int i = 0; i <= n; i++) {
+		for(int j = 0; j <= n; j++) {
 			dp[i][j] = MAX_N + 1;
 		}
 	}
 
-	for (int i = 0; i < n; i++) dp[i][i] = 1;
-	for (int s = 1; s <= n; s++) {
-		for (int i = 0; i <= n - s; i++) {
-			for (int j = i; j < i + s; j++) {
+	for(int i = 0; i < MAX_N; i++) dp[i][i] = 1;
+
+	for(int s = 1; s <= n; s++) {
+		for(int i = 0; i <= n - s; i++) {
+			for(int j = i; j < i + s; j++) {
 				int k = i + s - 1;
-				// explained in editorial
-				if (A[i] == A[k])
+				if(A[i] == A[k])
 					dp[i][k] = min(dp[i][k], dp[i][j] + dp[j + 1][k] - 1);
 				dp[i][k] = min(dp[i][k], dp[i][j] + dp[j + 1][k]);
 			}
 		}
 	}
+
 	cout << dp[0][n - 1] << endl;
 }
 ```
 
 </CPPSection>
 
+<JavaSection>
+	
+```java
+import java.io.*;
+import java.util.*;
+
+public class ModernArt3 {
+public static final int MAXN = 301;
+public static int[] inp = new int[MAXN];
+public static int[][] DP = new int[MAXN][MAXN];
```

Comment:
1. indentation is really weird
2. rename `inp` to like `painting` or smth
3. rename `DP` to `dp`

Reply:
nvm i fixed 1 myself